service/ui_plugin_build_baron.go

```diff
@@ -139,102 +181,135 @@ func (uis *UIServer) bbGetCreatedTickets(w http.ResponseWriter, r *http.Request)
 	gimlet.WriteJSON(w, results)
 }
 
-func (uis *UIServer) bbJiraSearch(rw http.ResponseWriter, r *http.Request) {
+// Retrieve the user feedback for a task id that has been stored by
+// the BF Suggestion Server.
+func (uis *UIServer) bbGetFeedback(rw http.ResponseWriter, r *http.Request) {
 	vars := gimlet.GetVars(r)
 	taskId := vars["task_id"]
 	exec := vars["execution"]
-	oldId := fmt.Sprintf("%v_%v", taskId, exec)
-	t, err := task.FindOneOld(task.ById(oldId))
+
+	// grab the user info
+	u := MustHaveUser(r)
+
+	t, bfsc, err := uis.bbGetTaskAndBFSuggestionClient(taskId, exec)
 	if err != nil {
 		gimlet.WriteJSONInternalError(rw, err.Error())
 		return
 	}
-	// if the archived task was not found, we must be looking for the most recent exec
-	if t == nil {
-		t, err = task.FindOne(task.ById(taskId))
-		if err != nil {
-			gimlet.WriteJSONInternalError(rw, err.Error())
-			return
-		}
-	}
 
-	bbProj, ok := uis.buildBaronProjects[t.Project]
-	if !ok {
-		gimlet.WriteJSON(rw, fmt.Sprintf("Corresponding JIRA project for %v not found", t.Project))
+	feedbackItems, err := bfsc.getFeedback(context.TODO(), t, u.Username())
+	if err != nil {
+		gimlet.WriteJSONInternalError(rw, err.Error())
 		return
 	}
+	gimlet.WriteJSON(rw, feedbackItems)
+}
 
-	fallback := &jiraSuggest{bbProj, uis.jiraHandler}
-	altEndpoint := &altEndpointSuggest{bbProj}
+func (uis *UIServer) bbSendFeedback(rw http.ResponseWriter, r *http.Request) {
+	// grab the user info
+	u := MustHaveUser(r)
 
-	var tickets []thirdparty.JiraTicket
-	if bbProj.AlternativeEndpointURL != "" {
-		tickets, err = raceSuggesters(fallback, altEndpoint, t)
-	} else {
-		tickets, err = fallback.Suggest(context.TODO(), t)
+	var input struct {
+		FeedbackType string                 `json:"type"`
+		FeedbackData map[string]interface{} `json:"data"`
+		TaskId       string                 `json:"task_id"`
+		Execution    int                    `json:"execution"`
 	}
 
-	jql := t.GetJQL(bbProj.TicketSearchProjects)
+	if err := json.NewDecoder(r.Body).Decode(&input); err != nil {
```

Comment:
I'd just use json.Unmarshal

Reply:
I updated this line to use `util.ReadJSONInto()`.